src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/DeleteManagedOrganizationUserAccountCommand.cs

```diff
@@ -50,140 +50,188 @@ public DeleteManagedOrganizationUserAccountCommand(
-                var orgUser = orgUsers.FirstOrDefault(ou => ou.Id == orgUserId);
+                orgUser = orgUsers.FirstOrDefault(ou => ou.Id == orgUserId);
                 if (orgUser == null || orgUser.OrganizationId != organizationId)
                 {
                     throw new NotFoundException("Member not found.");
                 }
 
-                await ValidateDeleteUserAsync(organizationId, orgUser, deletingUserId, managementStatus, hasOtherConfirmedOwners);
+                user = users.FirstOrDefault(u => u.Id == orgUser.UserId);
 
-                var user = users.FirstOrDefault(u => u.Id == orgUser.UserId);
                 if (user == null)
                 {
                     throw new NotFoundException("Member not found.");
                 }
 
-                await ValidateUserMembershipAndPremiumAsync(user);
+                await ValidateDeleteUserAsync(organizationId, orgUser, user, deletingUserId, managementStatus, hasOtherConfirmedOwners);
 
-                results.Add((orgUserId, string.Empty));
+                await CancelPremiumAsync(user);
+
+                userDeletionResults.Add((orgUserId, orgUser, user, string.Empty));
             }
             catch (Exception ex)
             {
-                results.Add((orgUserId, ex.Message));
+                userDeletionResults.Add((orgUserId, orgUser, user, ex.Message));
             }
         }
 
-        var orgUserResultsToDelete = results.Where(result => string.IsNullOrEmpty(result.ErrorMessage));
-        var orgUsersToDelete = orgUsers.Where(orgUser => orgUserResultsToDelete.Any(result => orgUser.Id == result.OrganizationUserId));
-        var usersToDelete = users.Where(user => orgUsersToDelete.Any(orgUser => orgUser.UserId == user.Id));
+        await HandleUserDeletionsAsync(userDeletionResults);
 
-        if (usersToDelete.Any())
-        {
-            await DeleteManyAsync(usersToDelete);
-        }
+        await LogDeletedOrganizationUsersAsync(userDeletionResults);
+
+        return userDeletionResults
+            .Select(i => (i.OrganizationUserId, i.ErrorMessage))
+            .ToList();
+    }
+
+    private async Task<IEnumerable<User>> GetUsersAsync(ICollection<OrganizationUser> orgUsers)
+    {
+        var userIds = orgUsers
+         .Where(orgUser => orgUser.UserId.HasValue)
+         .Select(orgUser => orgUser.UserId!.Value)
+         .ToList();
 
-        await LogDeletedOrganizationUsersAsync(orgUsers, results);
+        var users = await _userRepository.GetManyAsync(userIds);
 
-        return results;
+        return users;
     }
 
-    private async Task ValidateDeleteUserAsync(Guid organizationId, OrganizationUser orgUser, Guid? deletingUserId, IDictionary<Guid, bool> managementStatus, bool hasOtherConfirmedOwners)
+    private async Task ValidateDeleteUserAsync(Guid organizationId, OrganizationUser orgUser, User user, Guid? deletingUserId, IDictionary<Guid, bool> managementStatus, bool hasOtherConfirmedOwners)
```

Comment:
Redundant naming: we wouldn't be validating anything else here.

```suggestion
    private async Task ValidateAsync(Guid organizationId, OrganizationUser orgUser, User user, Guid? deletingUserId, IDictionary<Guid, bool> managementStatus, bool hasOtherConfirmedOwners)
```